src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,30 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MINIMUM_AGE = 18;
+    private static final int MINIMUM_LOGIN_LENGTH = 6;
+    private static final int MINIMUM_PASSWORD_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new ValidationException("LOGIN " + user.getLogin() + "is used");
+        }
+
+        if (user.getPassword().length() < MINIMUM_PASSWORD_LENGTH
+                    || user.getPassword() == null) {
+            throw new ValidationException("PASSWORD " + user.getPassword() + "is to short");
```

Comment:
Message says that password is too short, but if password is null you don't provide correct exception message.